stripe/src/main/java/com/stripe/android/payments/DefaultStripeChallengeStatusReceiver.kt

```diff
@@ -0,0 +1,259 @@
+package com.stripe.android.payments
+
+import com.stripe.android.AnalyticsEvent
+import com.stripe.android.Logger
+import com.stripe.android.StripeIntentResult
+import com.stripe.android.exception.StripeException
+import com.stripe.android.model.StripeIntent
+import com.stripe.android.networking.AnalyticsDataFactory
+import com.stripe.android.networking.AnalyticsRequest
+import com.stripe.android.networking.AnalyticsRequestExecutor
+import com.stripe.android.networking.ApiRequest
+import com.stripe.android.networking.RetryDelaySupplier
+import com.stripe.android.networking.StripeRepository
+import com.stripe.android.stripe3ds2.transaction.ChallengeFlowOutcome
+import com.stripe.android.stripe3ds2.transaction.CompletionEvent
+import com.stripe.android.stripe3ds2.transaction.ProtocolErrorEvent
+import com.stripe.android.stripe3ds2.transaction.RuntimeErrorEvent
+import com.stripe.android.stripe3ds2.transaction.StripeChallengeStatusReceiver
+import com.stripe.android.stripe3ds2.transaction.Transaction
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import kotlin.coroutines.CoroutineContext
+
+internal class DefaultStripeChallengeStatusReceiver internal constructor(
```

Comment:
Is there only a challenge status receiver for the 3DS?   It might be helpful to have that in the name if it is.

Reply:
It's a 3DS2 SDK thing so I'm not worried about it being confusing.